proxy/modules/native_module_host/src/native_module_host.c

```diff
@@ -0,0 +1,418 @@
+#include "module.h"
+#include "module_access.h"
+#include "module_loaders/dynamic_loader.h"
+#include "native_module_host.h"
+#include "parson.h"
+
+
+#define LOADER_NAME_KEY "name"
+#define LOADER_ENTRYPOINT_KEY "entrypoint"
+
+
+typedef struct MODULE_HOST_TAG
+{
+	MODULE_LIBRARY_HANDLE module_library_handle;
+	const MODULE_LOADER* module_loader;
+	MODULE_HANDLE module;
+	BROKER_HANDLE module_host_broker;
+} MODULE_HOST;
+
+// Arguments coming to this module are the module_args coming from the _Create_ control message.
+// in this circumstance, all we are goint to do is copy the string to send to our Module_Create function.
+static void* NativeModuleHost_ParseConfigurationFromJson(const char* configuration)
+{
+	char* config_str;
+	if (configuration == NULL)
+	{
+		/*Codes_SRS_NATIVEMODULEHOST_17_003: [ NativeModuleHost_ParseConfigurationFromJson shall return NULL if configuration is NULL. ]*/
+		config_str = NULL;
+	}
+	else
+	{
+		/*Codes_SRS_NATIVEMODULEHOST_17_004: [ On success, NativeModuleHost_ParseConfigurationFromJson shall return a valid copy of the configuration string given. ]*/
+		if (mallocAndStrcpy_s(&config_str, configuration) != 0)
+		{
+			/*Codes_SRS_NATIVEMODULEHOST_17_005: [ NativeModuleHost_ParseConfigurationFromJson shall return NULL if any system call fails. ]*/
+			config_str = NULL;
+		}
+	}
+    return config_str;
+}
+
+// Free the string we copied.
+static void NativeModuleHost_FreeConfiguration(void* configuration)
+{
+	/*Codes_SRS_NATIVEMODULEHOST_17_006: [ NativeModuleHost_FreeConfiguration shall do nothing if passed a NULL configuration. ]*/
+    if (configuration != NULL)
+    {
+		/*Codes_SRS_NATIVEMODULEHOST_17_007: [ NativeModuleHost_FreeConfiguration shall free configuration. ]*/
+        free((char*)configuration);
+    }
+}
+
+static int parse_loader(JSON_Object* loader_json, GATEWAY_MODULE_LOADER_INFO* loader_info)
+{
+    int result;
+
+    // get loader name; we assume it is "native" if the JSON doesn't have a name
+	/*Codes_SRS_NATIVEMODULEHOST_17_013: [ NativeModuleHost_Create shall get "name" string from the "outprocess.loader" object. ]*/
+    const char* loader_name = json_object_get_string(loader_json, LOADER_NAME_KEY);
+    if (loader_name == NULL)
+    {
+		/*Codes_SRS_NATIVEMODULEHOST_17_014: [ If the loader name string is not present, then NativeModuleHost_Create shall assume the loader is the native loader name. ]*/
+        loader_name = DYNAMIC_LOADER_NAME;
+    }
+
+    // locate the loader
+	/*Codes_SRS_NATIVEMODULEHOST_17_015: [ NativeModuleHost_Create shall find the module loader by name. ]*/
+    const MODULE_LOADER* loader = ModuleLoader_FindByName(loader_name);
+    if (loader == NULL)
+    {
+		/*Codes_SRS_NATIVEMODULEHOST_17_026: [ If any step above fails, then NativeModuleHost_Create shall free all resources allocated and return NULL. ]*/
+		LogError("Loader JSON has a non-existent loader 'name' specified - %s.", loader_name);
+        result = 1;
```

Comment:
>       result = 1; [](start = 1, length = 18)

What are these numbers? I don't see a documentation of these errors. We shall keep consistent and log __LINE__ or document these numbers shall it means something...